src/twisted/trial/_dist/test/test_workerreporter.py

```diff
@@ -88,6 +90,30 @@ def test_addFailureTuple(self):
                          managercommands.AddFailure)
 
 
+    def test_addFailureNonASCII(self):
+        """
+        L{WorkerReporter.addFailure} sends a L{managercommands.AddFailure}
+        message when called with a L{Failure}, even if it includes encoded
+        non-ASCII content.
+        """
+        content = u"\N{SNOWMAN}".encode("utf-8")
```

Comment:
I don't understand the test.

I was expecting to see `content = u"\N{SNOWMAN}"` , that is have an exception which has Unicode as args[0]

Reply:
The failure that I observe comes from having bytes that can't be decoded as ASCII.  So this requires bytes here, not unicode.